### ISSUE TITLE
Make sure that GOV.UK Frontend meets WCAG 2.1	

### ISSUE THREAD

**Issue by timpaul** [alphagov/govuk-design-system#677]
## What

Make sure that the Design System, and everything in it meets at least level AA of the WCAG 2.1 guidelines.

## Why

All services that went live after 23 September 2018 must meet WCAG2.1 AA by 23 September 2019. Older services must meet the same standard by 23 September 2020. We need to support service teams by ensuring that everything in the GOV.UK Design System meets this standard.

#### Must

- [x] [Run aXe on the GOV.UK Design System](https://github.com/alphagov/govuk-design-system/issues/620)
- [x] [Investigate colour contrast issues with the focus state](https://github.com/alphagov/govuk-frontend/issues/1137)
- [x] [Investigate colour contrast issues with hint text](https://github.com/alphagov/govuk-frontend/issues/1134)
- [x] [Investigate colour contrast issues within the footer component](https://github.com/alphagov/govuk-frontend/issues/1135)
- [x] [Investigate colour contrast issues with the header component](https://github.com/alphagov/govuk-frontend/issues/1133)
- [x] [Investigate colour contrast issues within the panel component](https://github.com/alphagov/govuk-frontend/issues/1132)
- [x] [Link hover state does not meet contrast requirements on font sizes below 18px](https://github.com/alphagov/govuk-frontend/issues/987)
- [x] [Tab contrast](https://github.com/alphagov/govuk-design-system-backlog/issues/100)
- [x] [Publish guidance on meeting colour contrast](https://github.com/alphagov/design-system-team-internal/issues/59)
- [x] [Make it easy to set autocomplete attributes on form inputs](https://github.com/alphagov/govuk-frontend/issues/1136)
- [x] [Check that the Design System works with adaptable text](https://github.com/alphagov/govuk-design-system/issues/843)
- [x] [Update the Design System page templates](https://github.com/alphagov/govuk-design-system/issues/699)

#### Should
[Better align custom tab logic with tab component](https://github.com/alphagov/govuk-design-system/issues/385)
Publish accessibility principles, criteria and guidance
Investigate potential issues with conditional reveals ([#979](https://github.com/alphagov/govuk-frontend/issues/979), [#935](https://github.com/alphagov/govuk-frontend/issues/935), [#989](https://github.com/alphagov/govuk-frontend/issues/989))

#### Could
[Content below buttons on start pages](https://github.com/alphagov/govuk-design-system-backlog/issues/111#issuecomment-364082973)
[Investigate whether we should specify 'aria-invalid' on form inputs that have errors](https://github.com/alphagov/govuk-frontend/issues/1168)
[Investigate adding iOS dynamic type support](https://github.com/alphagov/govuk-frontend/issues/882)

### 2. Modify the components and test
- [x] [Change review app to include code from legacy repos](https://github.com/alphagov/govuk-frontend/issues/1262)
- [x] [Update links (and things that look like links) to use the new focus style #1302](https://github.com/alphagov/govuk-frontend/issues/1302)
- [x] [Update radios and checkboxes to use the new focus state #1304](https://github.com/alphagov/govuk-frontend/issues/1304)
- [x] [Update form inputs to use the new focus state #1303](https://github.com/alphagov/govuk-frontend/issues/1303)
- [x] [Update buttons to use the new styles and the new focus state #1305](https://github.com/alphagov/govuk-frontend/issues/1305)
- [x] [Update tabs to use the new styles and the new focus state #1306](https://github.com/alphagov/govuk-frontend/issues/1306)
- [x] [Update accordion to use the new styles and the new focus state #1307](https://github.com/alphagov/govuk-frontend/issues/1307)
- [x] [Update footer to use the new focus states #1320](https://github.com/alphagov/govuk-frontend/issues/1320)
- [x] Test with assistive technologies
- [x] [Decide on how the mixins should change for new focus styles](https://github.com/alphagov/govuk-frontend/issues/1344)

### 3. External audit of the Design System
- [x] [Update DAC brief before audit](https://github.com/alphagov/design-system-team-internal/issues/95)
- [x] [Send brief to potential audit suppliers](https://github.com/alphagov/design-system-team-internal/issues/24)
- [x] [Develop the test pages for the accessibility audit](https://github.com/alphagov/design-system-team-internal/issues/23)
- [x] [Arrange a call with DAC to check requirements](https://github.com/alphagov/design-system-team-internal/issues/57)
- [x] [Build an example page of smaller checkboxes and radio buttons for the accessibility audit](https://github.com/alphagov/design-system-team-internal/issues/61)
- [x] [Build an example page of secondary and destructive buttons for the accessibility audit](https://github.com/alphagov/design-system-team-internal/issues/62)
- [x] Run the accessibility audit
- [x] Analyse results of the accessibility audit
- [x] Prioritise any remedial work for Q1

### 4. Prepare for publishing

See this epic - [Release Frontend 3.0](https://github.com/alphagov/govuk-design-system/issues/885)

### Notes
See 30th April team [catch up notes](https://docs.google.com/document/d/1tjvv6xYUD9bj-pujBnFMceEk2cSBd2QHxuil_we0qWs/)

**Comment by alex-page** [comment]
Oh yeah!!! We should collaborate ASAP. The whole reason our Design System exists is to meet WCAG 2.1. There is no benefit working seperately on these problems.

**Comment by alex-page** [comment]
Hey friends you have a lot of problems with contrast and you are using sass. Feel free to use our open source package [sass a11ycolor](https://github.com/alex-page/sass-a11ycolor) to find the closest accessible color using sass.

We use it to generate our custom colour schemes, you can see it working by going to our customise page here and entering in the [same colour value](https://designsystem.gov.au/templates/home/customise/?&text=red&action=red&background=red&textDark=red&actionDark=red&backgroundDark=red ). It makes some of the reds darker to make sure the contrast passes.

We use it in other ways like generating muted colours, we mix the background and foreground colours together then get the closest accessible colour of the mixed colour on the background. Have a look here:
https://github.com/govau/design-system-components/blob/da652cbab7ee3417812676314d1e1e5418f89059/packages/core/src/sass/_globals.scss#L807-L819

**Comment by nickcolley** [comment]
I have raised all the issues for the Design System website which brings us to 34 total GitHub issues which matches the DAC issue numbers that go to 33 (plus Ollie's in person issue).

https://github.com/issues?q=label%3A%22audit%3A+may-2019%22+user%3Aalphagov
